tests/unit/type/hash-field-expire.tcl

```diff
@@ -84,555 +89,1078 @@ proc hrandfieldTest {activeExpireConfig} {
+            r hpexpire h1 100 NX 1 01
+            r hpexpire h2 100 NX 1 01
+            r hpexpire h2 100 NX 1 02
+            r hpexpire h3 100 NX 1 01
+            r hpexpire h4 100 NX 1 2
+            r hpexpire h5 100 NX 1 3
+            r hpexpire h6 100 NX 1 05
+            r hpexpire h18 100 NX 17 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15 16 17
+
+            after 150
+
+            # Verify HDEL not ignore expired field. It is too much overhead to check
+            # if the field is expired before deletion.
+            assert_equal [r HDEL h1 01] "1"
+
+            # Verify HGET ignore expired field
+            assert_equal [r HGET h2 01] ""
+            assert_equal [r HGET h2 02] ""
+            assert_equal [r HGET h3 01] ""
+            assert_equal [r HGET h3 02] "02"
+            assert_equal [r HGET h3 03] "03"
+            # Verify HINCRBY ignore expired field
+            assert_equal [r HINCRBY h4 2 1] "1"
+            assert_equal [r HINCRBY h4 3 1] "100"
+            # Verify HSTRLEN ignore expired field
+            assert_equal [r HSTRLEN h5 3] "0"
+            assert_equal [r HSTRLEN h5 4] "4"
+            assert_equal [lsort [r HKEYS h6]] "01 02 03 04 06"
+            # Verify HEXISTS ignore expired field
+            assert_equal [r HEXISTS h18 07] "0"
+            assert_equal [r HEXISTS h18 18] "1"
+            # Verify HVALS ignore expired field
+            assert_equal [lsort [r HVALS h18]] "18"
+            # Restore to support active expire
+            r debug set-active-expire 1
         }
 
-        # Wait for active expire
-        wait_for_condition 50 20 { [r EXISTS same40] == 0 } else { fail "hash `same40` should be expired" }
+        test "A field with TTL overridden with another value (TTL discarded) ($type)" {
+            r del myhash
+            r hset myhash field1 value1 field2 value2 field3 value3
+            r hpexpire myhash 10000 NX 1 field1
+            r hpexpire myhash 1 NX 1 field2
+
+            # field2 TTL will be discarded
+            r hset myhash field2 value4
+            after 5
+            # Expected TTL will be discarded
+            assert_equal [r hget myhash field2] "value4"
+            assert_equal [r httl myhash 2 field2 field3] "$T_NO_EXPIRY $T_NO_EXPIRY"
+            assert_not_equal [r httl myhash 1 field1] "$T_NO_EXPIRY"
+        }
 
-        # Verify that all fields got expired and keys got deleted
-        foreach h $hash_sizes {
-            for {set i 1} {$i <= $h} {incr i} {
-                assert_equal 0 [r HEXISTS mix$h f$i]
+        test "Modify TTL of a field ($type)" {
+            r del myhash
+            r hset myhash field1 value1
+            r hpexpire myhash 200 NX 1 field1
+            r hpexpire myhash 1000 XX 1 field1
+            after 15
+            assert_equal [r hget myhash field1] "value1"
+            assert_range [r hpttl myhash 1 field1] 900 1000
```

Comment:
this test failed in https://github.com/sundb/redis/actions/runs/8897338398/job/24635771987.
1. we can't guarantee `myhash` doesn't expire between `r hpexpire myhash 200 NX 1 field1` and `r hpexpire myhash 1000 XX 1 field1`.
2. we can't guarantedd `after 15` will not pass more than 100ms.